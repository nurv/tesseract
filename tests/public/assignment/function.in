int fact (int n) { return (n == 0) ? 1 : n * fact(n - 1); }
fact(3)
int fact (int n) { if (n == 0) return 1; else return n * fact(n - 1); }
fact(4)
int fact (int n) { int r = 1; for(; n > 0; n--) r *= n; return r; }
fact(5)
int fact (int n) { int r = 1; while(n > 0) r *= n--; return r; }
fact(6)
int foo() { return 1; }
int bar() { return foo(); }
bar()
int foo() { return 2; }
foo()
bar()
