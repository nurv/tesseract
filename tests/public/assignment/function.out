> > 6
> > 24
> > 120
> > 720
> > > 1
> > 2
> 1
> 